app/javascript/components/Posts.jsx

```diff
@@ -1,33 +1,89 @@
 import React, { useState, useEffect } from "react";
-import { Link } from "react-router-dom";
 
 import Post from "./Post";
 
 const Posts = () => {
-  const [posts, setPosts] = useState([]);
+  const [allPosts, setAllPosts] = useState([]);
+  const [currentPost, setCurrentPost] = useState("");
+  const [reload, setReload] = useState(false); // A hook for activating the useEffect hook to update the index of posts
+  const [toastMessage, setToastMessage] = useState(""); // The message of the toasts that this page displays
+
+  // Submits the current post inside the shown text area to the index
+  //  - Doesn't post if the post has no characters in it or too many characters..\
+  //  - Submits the post to the db index if text is valid.
+  const submitPost = () => {
+    const token = document.querySelector('meta[name="csrf-token"]').content;
+    const toast = new bootstrap.Toast(document.getElementById('postToast'));
+
+    if (currentPost.length === 0) {
+      setToastMessage("Nothing to tweet!")
+      toast.show()
+      return
+    }
+
+    if (currentPost.length > 500) {
+      setToastMessage("Tweet must be under 500 characters!")
+      toast.show()
+      return
+    }
```

Comment:
It would be cool if you could move the validation checks into a new function. It'd be more easily readable if there was a function called `checkPostValidations` or something like that.